### PR TITLE
[EventEngine] RunAfter migration: api_fuzzer

### PR DESCRIPTION
**Pull request by yijiem**

### CODE REVIEW

File: test/core/end2end/fuzzers/api_fuzzer.cc [hunk 1]
```diff
@@ -108,16 +107,15 @@ static std::atomic<bool> g_channel_force_delete{false};
...
 
 typedef struct addr_req {
-  grpc_timer timer;
   char* addr;
   grpc_closure* on_done;
   std::unique_ptr<grpc_core::ServerAddressList>* addresses;
 } addr_req;
 
-static void finish_resolve(void* arg, grpc_error_handle error) {
+static void finish_resolve(void* arg) {
```
Comment (drfloob) [inline test/core/end2end/fuzzers/api_fuzzer.cc:115]: `void* arg` can now just be `addr_req* r`. (or see suggestion below for makign this an `addr_req&& r` argument)

File: test/core/end2end/fuzzers/api_fuzzer.cc [hunk 2]
```diff
@@ -260,18 +259,14 @@ static void sched_connect(grpc_closure* closure, grpc_endpoint** ep,
...
 
 typedef struct {
-  grpc_timer timer;
   grpc_closure* closure;
   grpc_endpoint** ep;
   gpr_timespec deadline;
 } future_connect;
 
-static void do_connect(void* arg, grpc_error_handle error) {
+static void do_connect(void* arg) {
```
Comment (drfloob) [inline test/core/end2end/fuzzers/api_fuzzer.cc:267]: This also does not need to be a `void*` anymore.

File: test/core/end2end/fuzzers/api_fuzzer.cc [hunk 3]
```diff
@@ -308,9 +303,11 @@ static void sched_connect(grpc_closure* closure, grpc_endpoint** ep,
   fc->closure = closure;
   fc->ep = ep;
   fc->deadline = deadline;
-  grpc_timer_init(
-      &fc->timer, grpc_core::Duration::Seconds(1) + grpc_core::Timestamp::Now(),
-      GRPC_CLOSURE_CREATE(do_connect, fc, grpc_schedule_on_exec_ctx));
+  GetDefaultEventEngine()->RunAfter(grpc_core::Duration::Seconds(1), [fc] {
```
Comment (drfloob) [inline test/core/end2end/fuzzers/api_fuzzer.cc:306]: I think maybe `fc` can be a local variable now, no allocations required.

```
future_connect fc;
fc.closure = closure;
...
GetDefaultEventEngine()->RunAfter(grpc_core::Duration::Seconds(1), [fc = std::move(fc)] mutable {
    grpc_core::ApplicationCallbackExecCtx callback_exec_ctx;
    grpc_core::ExecCtx exec_ctx;
    do_connect(std::move(fc));
  });
```

File: test/core/end2end/fuzzers/api_fuzzer.cc [hunk 4]
```diff
@@ -244,9 +241,11 @@ grpc_ares_request* my_dns_lookup_ares(
   r->addr = gpr_strdup(addr);
   r->on_done = on_done;
   r->addresses = addresses;
-  grpc_timer_init(
-      &r->timer, grpc_core::Duration::Seconds(1) + grpc_core::Timestamp::Now(),
-      GRPC_CLOSURE_CREATE(finish_resolve, r, grpc_schedule_on_exec_ctx));
+  GetDefaultEventEngine()->RunAfter(grpc_core::Duration::Seconds(1), [r] {
```
Comment (drfloob) [inline test/core/end2end/fuzzers/api_fuzzer.cc:244]: See below for my suggestion about avoiding allocations, I think that can apply here too.

**Review by drfloob — COMMENTED** [review]
This is a nice one, no cancellation semantics to worry about.

File: test/core/end2end/fuzzers/api_fuzzer.cc (hunk 3 above)
Comment (yijiem) [reply test/core/end2end/fuzzers/api_fuzzer.cc]: Great point! Thanks for catching it! Looks like `future_connect` and `addr_req` are trivially-copyable types so no need to use `std::move` here.

File: test/core/end2end/fuzzers/api_fuzzer.cc (hunk 4 above)
Comment (yijiem) [reply test/core/end2end/fuzzers/api_fuzzer.cc]: Done.

File: test/core/end2end/fuzzers/api_fuzzer.cc (hunk 2 above)
Comment (yijiem) [reply test/core/end2end/fuzzers/api_fuzzer.cc]: Done.

File: test/core/end2end/fuzzers/api_fuzzer.cc (hunk 1 above)
Comment (yijiem) [reply test/core/end2end/fuzzers/api_fuzzer.cc]: Used `(addr_req r)` instead but thanks for the great catch!

**Comment by yijiem** [comment]
Thank you AJ for the review!

**Review by drfloob — APPROVED** [review]

**Pull request merged**